mesonbuild/compilers/detect.py

```diff
@@ -1220,9 +1220,16 @@ def detect_swift_compiler(env: 'Environment', for_machine: MachineChoice) -> Com
         # As for 5.0.1 swiftc *requires* a file to check the linker:
         with tempfile.NamedTemporaryFile(suffix='.swift') as f:
             cls = SwiftCompiler
-            linker = guess_nix_linker(env,
-                                      exelist, cls, version, for_machine,
-                                      extra_args=[f.name])
+
+            outhandle, outfile = tempfile.mkstemp('.out')
+            os.close(outhandle)
```

Comment:
Is it not possible to use `NamedTemporaryFile` here, to avoid the manual cleanup?